### PR TITLE
Merge for v0.2 release

### PR DESCRIPTION
**Pull request by erickpeirson**

### CODE REVIEW

File: fulltext/extract.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    logger.info('Retrieving PDF for %s' % document_id)
+    start_time = datetime.now()
+    try:
+        # Retrieve PDF from arXiv central document store.
+        pdf_path = retrievePDF.session.retrieve(pdf_url, document_id)
+        if pdf_path is None:
+            metrics.session.report('PDFIsAvailable', 0.)
+            msg = '%s: no PDF available' % document_id
+            logger.info(msg)
+            raise RuntimeError(msg)
```
Comment (rcplane) [inline fulltext/extract.py]: Would a FileNotFoundError or custom exception be more informative if we like this exception control flow pattern?

File: extractor/fulltext/launch_single.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+from fulltext import convert
+
+log = logger.getLogger('fulltext')
+
+if __name__ == '__main__':
+    if len(sys.argv) <= 1:
+        sys.exit('No file path specified')
+    path = sys.argv[1].strip()
+    try:
+        sys.stdout.write('Path: %s\n' % path)
```
Comment (rcplane) [inline extractor/fulltext/launch_single.py]: Should these lines also go to the logger?

File: fulltext/extract.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        logger.info('%s: retrieved PDF' % document_id)
+
+        logger.info('Attempting text extraction for %s' % document_id)
+        txt_path = extractor.session.extract_fulltext(pdf_path)
+        logger.info('Text extraction for %s succeeded with %s' %
+                    (document_id, txt_path))
+
+        os.remove(pdf_path)    # Cleanup.
+        with open(txt_path, encoding='utf-8') as f:
+            content = f.read()
```
Comment (rcplane) [inline fulltext/extract.py]: Do we know that reading the whole file will fit safely in memory?

Comment (erickpeirson) [reply fulltext/extract.py]: That's a great question; good to think through. 

Academic papers tend to be in the several- to dozens-of-pages range, but let's suppose that someone uploads a book manuscript to arXiv that is 500 pages long. A single-spaced page at 12pt font might have 500 words; assuming an average word length of 5 characters, that's 2,500 characters per page. 2,500 char/page * 500 pages = 1.25 million characters. In an extreme case, we might have an average of 2 bytes per character -- suppose someone making heavy use of accents or other non-ASCII symbols. So 2.5 megabytes. 

The bigger issue will be the PDF. Assume that we're generating PDFs that are 100kb/page (high, but suppose that there are figures) -- that's a 50MB PDF in our extreme example. Still a pretty reasonable number. 

As we get a bit further along with Kubernetes, we'll want to make these requirements more explicit.

File: extractor/fulltext/launch_single.py (hunk 2 above)
Comment (erickpeirson) [reply extractor/fulltext/launch_single.py]: Yep, good catch.

**Pull request merged**